### ISSUE TITLE
Conditional FastGAN

### ISSUE THREAD

**Issue by adambielski** [autonomousvision/projected_gan#10]
Has the conditional version of FastGAN been tested/should it show good results? I think in the paper there were no conditional experiments.
In my initial experiments it fails to learn anything meaningful; as a way to debug I also try to use the conditional architecture with just 1 class (so it should perform similarly to the unconditional one), but so far it also seems much worse.

In case of conditional training, I usually get
RuntimeWarning: overflow encountered in multiply
  img = (img - lo) * (255 / (hi - lo))

which might suggest some large values in the generator's output.
I'm trying now applying tanh to the output in the conditional case (that is what BigGAN does which also uses cond batchnorm), ~~so far the 1-cond learning curves look more similar to the unconditional case, but I only just started it~~

**Comment by xl-sr** [comment]
Yes, I tested it, otherwise, we wouldn't have included it ;)
What dataset are you trying currently?

**Comment by adambielski** [comment]
@xl-sr Right, sorry, I didn't meant to imply anything :) What I meant was - what kind of results can be expected with the conditional FastGAN / on what kind of datasets.
I was testing on my own small dataset and noticed much better results in the unconditional case, but it was a rather peculiar setting I was trying. I can't test it more right now so I'll be closing it for the moment.

It's a great work, thanks for sharing the code and assisting here!

**Comment by xl-sr** [comment]
haha, don't worry. :) I tried ImageNet and the results looked decent.
Let me know if you figure out the problem, as you said, the sanity check with a single class should be as good as the unconditional setting. But I am also not a huge fan of conditioning via batch norm, so maybe you will have more success by trying something different, e.g. conditioning via a mapping network (check StyleGAN2-ADA on Cifar10).

**Comment by adambielski** [comment]
Yeah I tried with StyleGAN2-ADA and it was decent (but I also tried it out before so no surprise there :))
